lib/read-write.js

```diff
@@ -9,32 +9,31 @@ const {
 
 function getRows(rows, options){
     return new Promise((resolve, reject) => {
-        if (!rows || !rows.length){reject("Need valid rows")}
-        if (!options){reject("Need valid options")}
-        if (!options.spreadsheetId || typeof options.privateKey !== "string"){reject("Need valid spreadsheetId")}
-        if (!options.clientEmail || typeof options.privateKey !== "string"){reject("Need valid clientEmail")}
-        if (!options.privateKey || typeof options.privateKey !== "string"){reject("Need valid privateKey")}
+        if (!rows || !rows.length){return reject(new Error("Need valid rows"))}
+        if (!options){return reject(new Error("Need valid options"))}
+        if (!options.spreadsheetId || typeof options.privateKey !== "string"){return reject(new Error("Need valid spreadsheetId"))}
+        if (!options.clientEmail || typeof options.privateKey !== "string"){return reject(new Error("Need valid clientEmail"))}
+        if (!options.privateKey || typeof options.privateKey !== "string"){return reject(new Error("Need valid privateKey"))}
 
         _sheets(options.clientEmail, options.privateKey).batchGet({
             spreadsheetId: options.spreadsheetId,
             ranges: _getRanges(rows),
             majorDimension: "ROWS",
             dateTimeRenderOption: options.dateTimeRenderOption || "FORMATTED_STRING"
         }, (error, response) => {
-            if (error){reject(error);}
+            if (error){return reject(error);}
             resolve(response);
         });
-    }).then(_sheetsToMappedObject.bind(null, rows))
-    .catch(console.error);
```

Comment:
Using `.catch()` to handle errors *in library code* short-circuits the ability of calling code to know anything of the error (including the ability to react to it).

The *most important* concept is: `.catch()` will handle return values much like `.then()` - you could (should) re-throw errors if library code must log them. 
Another powerful feature: let's say the error isn't serious, and you want to allow calling code to keep `.then`ing right along... Simply return a placeholder/dummy object and voilà, the chain keeps going.

That said, there's still a time for `.catch(console.error)` - usually at the top-most layer of your promise chain (like in tests - not express however).